v2/processor.go

```diff
@@ -25,6 +26,18 @@ type MessageSettler interface {
 	RenewMessageLock(ctx context.Context, message *azservicebus.ReceivedMessage, options *azservicebus.RenewMessageLockOptions) error
 }
 
+type ReceiverImpl struct { // shuttle.Receiver is already an exported interface
```

Comment:
Keep this struct private, an dname it just `receiver`.
add a constructor: 

shuttle.NamedReceiver(receiver, name) Receiver {
}

add Name() string to the Receiver interface we have, and always wrap the incoming receiver from the single receiver processor, with a default name.

Then you only depend on the interface.

WDYT?

Reply:
If you prefer to keep the struct public, which is ok, I have an issue with the `Impl` suffix.

NamedReceiver is better, but I'm worried that it's too precise (what happens when we want to add more properties?)

ReceiverEx (Extended) is a little better than Impl IMO, because it tells it's extended, not that it's an implementation of an interface...

open to your suggestion/preference.